lib/controllers/login.js

```diff
@@ -33,10 +33,24 @@ module.exports = function(req, res) {
       options = {};
     }
 
+    var hasSocialProviders = false;
+
+    // Iterate all social providers and check if there are any enabled.
+    if (config.socialProviders) {
+      for (var name in config.socialProviders) {
+        var provider = config.socialProviders[name];
+        if (provider.enabled) {
+          hasSocialProviders = true;
+          break;
+        }
+      }
+    }
+
```

Comment:
I like to do this with filter:

```
var hasSocialProviders = Object.keys(config.socialProviders).filter(function(name){
  return config.socialProvider[name].enabled;
}).length > 0
```

I'm not attached to either way, just sharing.   Underscore also has some nice functions for making these one-liner operations.